test/Ocelot.UnitTests/DownstreamUrlCreator/DownstreamUrlCreatorMiddlewareTests.cs

```diff
@@ -611,6 +611,43 @@ public void Should_map_when_query_parameters_has_same_names_with_placeholder()
         ThenTheQueryStringIs($"?roleId={roleid}&{everything}");
     }
 
+    [Theory]
+    [Trait("Bug", "2116")]
+    [InlineData("api/debug()")] // no query
+    [InlineData("api/debug%28%29")] // debug()
```

Comment:
Hello @ggnaegi,
I would like to delay my approval until we have made a decision on the additional test case:
```suggestion
    [InlineData("api/debug()")] // no query
    [InlineData("api/debug%28%29")] // encoded debug()
    [InlineData("api/debug()?special=(\\,*,+,?,|,{,},[,],(,),^,$,.,#, ,)&2=(2)&3=[3]&4={4}")] // with query
```
The third test has revealed issues within the middleware logic, indicating that this test case is likely to fail. The `MergeQueryStringsWithoutDuplicateValues` method struggles to handle URLs containing query strings with special characters, such as `Regex` patterns and reserved URL specification characters. This issue became apparent in release 20.0.0 during the refactoring of the `MergeQueryStringsWithoutDuplicateValues` method, which aimed to integrate existing and new logic (for example OData filters in query parameters). Regrettably, the method fails to process special characters in parameter values.

Would you recommend addressing the design bottleneck within this bug fix, or would creating a new TODO task be a better approach to ensure the current release proceeds without delay?

Reply:
@raman-m the whole thing is a mess, @int0x81 who is new to ocelot said straight away that the whole implementation could be optimized. For now, I would suggest merging the current changes, refactorings are out of scope in my opinion. @int0x81 would be happy to create a follow up issue and PR.